#! /bin/sh /usr/share/dpatch/dpatch-run
## byterun_interp_ia64_fix.dpatch Sven LUther <luther@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Disabled not-really-needed ia64 cast which broke with gcc 4.0.

@DPATCH@
--- ocaml-3.08.3/byterun/interp.c.orig	2005-08-09 16:58:22.000000000 +0000
+++ ocaml-3.08.3/byterun/interp.c	2005-08-09 16:58:42.000000000 +0000
@@ -55,11 +55,7 @@
 #  ifdef DEBUG
 #    define Next goto next_instr
 #  else
-#    ifdef __ia64__
-#      define Next goto *(void *)(jumptbl_base + *((uint32 *) pc)++)
-#    else
-#      define Next goto *(void *)(jumptbl_base + *pc++)
-#    endif
+#    define Next goto *(void *)(jumptbl_base + *pc++)
 #  endif
 #else
 #  define Instruct(name) case name
